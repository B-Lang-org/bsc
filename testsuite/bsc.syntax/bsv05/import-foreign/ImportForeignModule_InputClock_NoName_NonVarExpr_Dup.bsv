module launderClock(Clock i, Clock o);
  return i;
endmodule

import "BVI" MOD =
module mkMod (Empty ifc) ;

   default_clock no_clock ;
   no_reset ;

   input_clock (CLK1) <- launderClock(noClock) ;

   input_clock (CLK2) <- launderClock(noClock) ;

endmodule

