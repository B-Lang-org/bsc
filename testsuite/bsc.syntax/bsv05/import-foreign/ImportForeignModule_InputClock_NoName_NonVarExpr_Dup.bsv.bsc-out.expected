checking package dependencies
=== parsed:
package ImportForeignModule_InputClock_NoName_NonVarExpr_Dup;
module launderClock#(Clock i)(Clock);
  i
endmodule: launderClock

module mkMod(Empty);
  let _bv1 <- launderClock(noClock);
  let _bv0 <- launderClock(noClock);
  Empty _m__;
  _m__ <- liftModule((unexpected) module verilog "MOD";
		     clock _clk__1(CLK1, {-inhigh-});
		     clock _clk__2(CLK2, {-inhigh-});
			 ((ClockArg _clk__1, _bv1), (ClockArg _clk__2, _bv0))
			 {
			 }
			 SchedInfo [] [] [] []
			 No combinational paths from inputs to outputs);
endmodule: mkMod

endpackage: ImportForeignModule_InputClock_NoName_NonVarExpr_Dup

-----

compiling ImportForeignModule_InputClock_NoName_NonVarExpr_Dup.bsv
=== parsed:
package ImportForeignModule_InputClock_NoName_NonVarExpr_Dup;
module launderClock#(Clock i)(Clock);
  i
endmodule: launderClock

module mkMod(Empty);
  let _bv1 <- launderClock(noClock);
  let _bv0 <- launderClock(noClock);
  Empty _m__;
  _m__ <- liftModule((unexpected) module verilog "MOD";
		     clock _clk__1(CLK1, {-inhigh-});
		     clock _clk__2(CLK2, {-inhigh-});
			 ((ClockArg _clk__1, _bv1), (ClockArg _clk__2, _bv0))
			 {
			 }
			 SchedInfo [] [] [] []
			 No combinational paths from inputs to outputs);
endmodule: mkMod

endpackage: ImportForeignModule_InputClock_NoName_NonVarExpr_Dup

-----

All packages are up to date.
